lib/bolt/plugin/aws.rb

```diff
@@ -0,0 +1,112 @@
+# frozen_string_literal: true
+
+require 'json'
+
+module Bolt
+  class Plugin
+    class Aws
+      class EC2
+        attr_accessor :client
+        attr_reader :config
+
+        def initialize(config)
+          require 'aws-sdk-ec2'
+          @config = config
+          @logger = Logging.logger[self]
+        end
+
+        def name
+          'aws::ec2'
+        end
+
+        def hooks
+          %w[inventory_targets]
+        end
+
+        def config_client(opts)
+          return client if client
+
+          options = {}
+
+          if opts.key?('region')
+            options[:region] = opts['region']
+          end
+          if opts.key?('profile')
+            options[:profile] = opts['profile']
+          end
+          if config.aws&.key?('credentials')
+            creds = File.expand_path(config.aws['credentials'])
+            if File.exist?(creds)
+              options[:credentials] = ::Aws::SharedCredentials.new(path: creds)
+            else
+              @logger.warn("Cannot load credentials file #{config.aws['credentials']}")
+            end
+          end
+
+          ::Aws::EC2::Client.new(options)
+        end
+
+        def inventory_targets(opts)
+          client = config_client(opts)
+          resource = ::Aws::EC2::Resource.new(client: client)
+
+          # Retrieve a list of EC2 instances and create a list of targets
+          # Note: It doesn't seem possible to filter stubbed responses...
+          resource.instances(filters: opts['filters']).map do |instance|
+            next unless instance.state.name == 'running'
+            target = {}
+
+            if opts.key?('uri')
+              uri = lookup(instance, opts['uri'])
```

Comment:
Would we ever want to lookup a default property for either `uri` or `name` if one wasn't provided in the inventory? For example, if the user did not specify a uri, then lookup would default to `public_ip_address`?

On one hand this simplifies things if the user doesn't know what properties are available / want less set up to get the plugin working... on the other hand it seems kind of like 'magic'.